pyretailscience/plots/line.py

```diff
@@ -0,0 +1,317 @@
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        source_text (str, optional): The source text to add to the plot.
+        move_legend_outside (bool, optional): Move the legend outside the plot.
+        **kwargs: Additional keyword arguments for Pandas' `plot` function.
+
+    Returns:
+        SubplotBase: The matplotlib axes object.
+    """
+    if x_col is not None:
+        _check_datetime_column(df, x_col)
+
+    if group_col is not None:
+        unique_groups = df[group_col].unique()
+        colors = [
+                     COLORS[color][shade]
+                     for shade in [500, 300, 700]
+                     for color in ["green", "orange", "red", "blue", "yellow", "violet", "pink"]
+                 ][: len(unique_groups)]
+
+        ax = None
+        for i, group in enumerate(unique_groups):
+            group_df = df[df[group_col] == group]
+
+            if x_col is not None:
+                ax = group_df.plot(
+                    x=x_col,
+                    y=value_col,
+                    ax=ax,
+                    linewidth=3,
+                    color=colors[i],
+                    label=group,
+                    legend=True,
+                    **kwargs,
+                )
+            else:
+                ax = group_df.plot(
+                    y=value_col,
+                    ax=ax,
+                    linewidth=3,
+                    color=colors[i],
+                    label=group,
+                    legend=True,
+                    **kwargs,
+                )
+    else:
+        colors = COLORS["green"][500]
+        if x_col is not None:
+            ax = df.plot(
+                x=x_col,
+                y=value_col,
+                linewidth=3,
+                color=colors,
+                legend=False,
+                ax=ax,
+                **kwargs,
+            )
+        else:
+            ax = df.plot(
+                y=value_col,
+                linewidth=3,
+                color=colors,
+                legend=False,
+                ax=ax,
+                **kwargs,
+            )
+
+    ax = gu.standard_graph_styles(
+        ax,
+        title=title if title else f"{value_col.title()} Over {x_col.title()}" if x_col else f"{value_col.title()} Over Index",
+        x_label=x_label if x_label else (x_col.title() if x_col else "Index"),
+        y_label=y_label or value_col.title(),
+    )
+
+    if move_legend_outside:
+        ax.legend(bbox_to_anchor=(1.05, 1))
+
+    if legend_title is not None:
+        ax.legend(title=legend_title)
+
+    if source_text:
+        ax.annotate(
+            source_text,
+            xy=(-0.1, -0.2),
+            xycoords="axes fraction",
+            ha="left",
+            va="center",
+            fontsize=GraphStyles.DEFAULT_SOURCE_FONT_SIZE,
+            fontproperties=GraphStyles.POPPINS_LIGHT_ITALIC,
+            color="dimgray",
+        )
+
+    for tick in ax.get_xticklabels() + ax.get_yticklabels():
```

Comment:
I think this can be replaced with ` gu.standard_tick_styles(ax)`. See [here](https://github.com/Data-Simply/pyretailscience/blob/main/pyretailscience/standard_graphs.py#L441C4-L441C32)